ContextualPopupDecorator/ContextualPopupDecorator.js

```diff
@@ -0,0 +1,734 @@
+			this.updateLeaveFor(current);
+			this.setState({
+				activator: current
+			});
+			this.spotPopupContent();
+		}
+
+		handleClose = () => {
+			this.updateLeaveFor(null);
+			this.setState({
+				activator: null
+			});
+		}
+
+		handleDirectionalKey (ev) {
+			// prevent default page scrolling
+			ev.preventDefault();
+			// stop propagation to prevent default spotlight behavior
+			ev.stopPropagation();
+			// set the pointer mode to false on keydown
+			Spotlight.setPointerMode(false);
+		}
+
+		// handle key event from outside (i.e. the activator) to the popup container
+		handleKeyDown = (ev) => {
+			const {activator, containerId} = this.state;
+			const {spotlightRestrict} = this.props;
+			const current = Spotlight.getCurrent();
+			const direction = getDirection(ev.keyCode);
+
+			if (!direction) return;
+
+			const hasSpottables = Spotlight.getSpottableDescendants(containerId).length > 0;
+			const spotlessSpotlightModal = spotlightRestrict === 'self-only' && !hasSpottables;
+			const shouldSpotPopup = current === activator && direction === this.adjustedDirection && hasSpottables;
+
+			if (shouldSpotPopup || spotlessSpotlightModal) {
+				this.handleDirectionalKey(ev);
+
+				// we guard against attempting a focus change by verifying the case where a
+				// spotlightModal popup contains no spottable components
+				if (!spotlessSpotlightModal && shouldSpotPopup) {
+					this.spotPopupContent();
+				}
+			}
+		}
+
+		// handle key event from contextual popup and closes the popup
+		handleContainerKeyDown = (ev) => {
+			// Note: Container will be only rendered if `open`ed, therefore no need to check for `open`
+			const direction = getDirection(ev.keyCode);
+
+			if (!direction) return;
+
+			this.handleDirectionalKey(ev);
+
+			// if focus moves outside the popup's container, issue the `onClose` event
+			if (Spotlight.move(direction) && !this.containerNode.contains(Spotlight.getCurrent())) {
+				forward('onClose', ev, this.props);
+			}
+		}
+
+		spotActivator = (activator) => {
+			if (!Spotlight.getPointerMode() && activator && activator === Spotlight.getCurrent()) {
+				activator.blur();
+			}
+			if (!Spotlight.focus(activator)) {
+				Spotlight.focus();
+			}
+		}
+
+		spotPopupContent = () => {
+			const {spotlightRestrict} = this.props;
+			const {containerId} = this.state;
+			const spottableDescendants = Spotlight.getSpottableDescendants(containerId);
+			if (spotlightRestrict === 'self-only' && spottableDescendants.length && Spotlight.getCurrent()) {
+				Spotlight.getCurrent().blur();
+			}
+
+			if (!Spotlight.focus(containerId)) {
+				Spotlight.setActiveContainer(containerId);
+			}
+		}
+
+		render () {
+			const {'data-webos-voice-exclusive': voiceExclusive, showCloseButton, popupComponent: PopupComponent, popupClassName, noAutoDismiss, onClose, open, popupProps, skin, spotlightRestrict, ...rest} = this.props;
+			const scrimType = spotlightRestrict === 'self-only' ? 'transparent' : 'none';
```

Comment:
We might not support `scrimType` feature because its prop was not defined. If correct, please remove the code related with the `scrimType`.